### ISSUE TITLE
Entry point in static scope requires [Export] after first compile in new project

### ISSUE THREAD

**Issue by EnokViking** [beefytech/Beef#1008]
In 0.43.0, compiling the following produces the desired result:
```cs
namespace TestProj 
{
    static {
        //[Export]
        public static void Main() {
            Console.WriteLine("Test");
        }
    }
}
```
However, if you were to add a new console print and compile it again, you get an error about an unreferenced external symbol, regarding Main() (which can be solved by uncommenting [Export]

**Comment by EnokViking** [comment]
This only seems to happen when there's a startup object defined. If I just remove that and move the main method into the global namespace, it works as expected.

**Comment by bfiete** [comment]
Fixed at df02aa3b6a5ce3263f23f61fdf6601910aca1a52
